RIPS/rip-7560.md

```diff
@@ -276,14 +275,12 @@ The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with em
 The frame must return a bytes array that is interpreted as:
 
 ```solidity
-
-abi.encode(context, MAGIC_VALUE_PAYMASTER, validUntil, validAfter)
-
+MAGIC_VALUE_PAYMASTER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>, 64 <constant, 32 bytes>, length <32 bytes>, context <variable>
```

Comment:
I think at this point there is no benefit in "truncating" the magic value to 20 bytes (because we expect an address there sometime in the future) and we should just give it 32 bytes. Makes it much easier to return the values in Solidity.

Reply:
at this point all wallets are 4337, which means they already do it.
we either `returns( bytes32 validationData )` (which is a packing of after/until/magic)
or  `returns (uint256 magic, uint256 validAfter, uint256 validUntil)`
(and an extra `bytes context` for paymaster)

Reply:
>`returns (uint256 magic, uint256 validAfter, uint256 validUntil)` (and an extra `bytes context` for paymaster)

I think this would be much better and won't actually cost more than the packed version. I strongly suggest we consider doing exactly this and returning each variable with a native 256-bit value instead of shifting, packing and so on.